sources/api/netdog/src/networkd_status.rs

```diff
@@ -0,0 +1,304 @@
+//! The networkd_status module contains definitions and functions for tracking network status from networkd
+//!
+use crate::interface_id::InterfaceName;
+#[cfg(net_backend = "systemd-networkd")]
+use crate::NETWORKCTL;
+use serde::{Deserialize, Deserializer};
+#[cfg(net_backend = "systemd-networkd")]
+use snafu::{ensure, ResultExt};
+#[cfg(net_backend = "systemd-networkd")]
+use std::cmp::Ordering;
+use std::convert::TryInto;
+use std::net::IpAddr;
+#[cfg(net_backend = "systemd-networkd")]
+use std::process::Command;
+
+#[derive(Clone, Debug, Deserialize)]
+#[serde(deny_unknown_fields)]
+#[allow(dead_code)]
+pub(crate) struct NetworkdDnsConfig {
+    #[serde(rename = "Family")]
+    family: u8,
+    #[serde(rename = "Address", deserialize_with = "ipaddr_from_vec_de")]
+    pub(crate) address: IpAddr,
+    #[serde(rename = "ConfigSource")]
+    config_source: String,
+    #[serde(rename = "ConfigProvider", deserialize_with = "ipaddr_from_vec_de")]
+    config_provider: IpAddr,
+}
+
+#[derive(Clone, Debug, Deserialize)]
+#[serde(deny_unknown_fields)]
+#[allow(dead_code)]
+pub(crate) struct SearchDomain {
+    #[serde(rename = "Domain")]
+    pub(crate) domain: String,
+    #[serde(rename = "ConfigSource")]
+    config_source: String,
+    #[serde(rename = "ConfigProvider", deserialize_with = "ipaddr_from_vec_de")]
+    config_provider: IpAddr,
+}
+
+#[derive(Debug, Deserialize)]
+pub(crate) struct NetworkctlIpAddr {}
```

Comment:
Is this used anywhere?  Maybe I'm just missing it.

Reply:
This is left over from the original way I was deserializing, its no longer needed and can be removed.